### ISSUE TITLE
Flatten and vec don't support the "order" argument

### ISSUE THREAD

**Issue by SteveDiamond** [cvxpy/cvxpy#1695]
**Describe the bug**
``expr.flatten()`` and ``vec(expr)`` should support the ``order`` argument, to choose C versus Fortran ordering.

**To Reproduce**
```
import cvxpy as cp

x = cp.Variable((3, 2))
expr = cp.vec(x, order='C')
expr = x.flatten(order='C')
```

**Expected behavior**
The behavior should be the same as ``reshape(expr, (expr.size), order=...)``.

**Output**
If applicable, include program output. If reporting a program crash, please include the entire stack trace.

**Version**
 - OS: Mac 10.14
 - CVXPY Version: 1.2

**Additional context**

**Comment by rileyjmurray** [comment]
Actually @SteveDiamond, I'm not sure if I'd classify this as a bug. Is there documentation that says that users can expect ``cp.vec`` or ``expr.flatten`` to accept an order argument? I ask because if there isn't, then the expected behavior you described would technically be a new feature, rather than a bugfix. This is relevant for deciding whether something can be updated in cvxpy 1.1.X and 1.2.X or whether it would need to released in 1.3.X

**Comment by SteveDiamond** [comment]
It's not a bug from that viewpoint.

**Comment by pat749** [comment]
in documentation there is nothing said about ,so it is a new feature and in (expr = cp.vec(x, order='C')) line vec() got an unexpected keyword argument 'order',so it is unevaluated expression and here it is not used in  colloquial expression so its new features or bug.

**Comment by rileyjmurray** [comment]
Resolved by #1754.
